addons/delivery/models/res_partner.py

```diff
@@ -9,9 +11,58 @@ class ResPartner(models.Model):
 
     property_delivery_carrier_id = fields.Many2one('delivery.carrier', company_dependent=True, string="Delivery Method", help="Used in sales orders.")
     is_pickup_location = fields.Boolean()  # Whether it is a pickup point address.
+    location_data = fields.Json(help="Information needed by shipping providers in case of pickup point addresses.")
 
     def _get_delivery_address_domain(self):
         return expression.AND([
             super()._get_delivery_address_domain(),
             [('is_pickup_location', '=', False)],
         ])
+
+    @api.model
+    def _address_from_json(self, json_location_data, parent_id, is_pickup_location=True):
+        """ Searches for an existing address with the same data as the one in json_location_data
+        and the same parent_id. If no address is found, creates a new one. """
+        location_data = json.loads(json_location_data)
+        if location_data:
+            name = location_data.get('name', False)
+            street = location_data.get('street', False)
+            city = location_data.get('city', False)
+            zip_code = location_data.get('zip_code', False)
+            country_code = location_data.get('country_code', False)
+            country = self.env['res.country'].search([('code', '=', country_code)]).id
+            state = self.env['res.country.state'].search([
+                ('code', '=', location_data.get('state', False)),
+                ('country_id', '=', country),
+            ]).id if (location_data.get('state') and country) else None
+            email = location_data.get('email', False)
+            phone = location_data.get('phone', False)
+
+            domain = [
+                ('name', '=', name),
+                ('street', '=', street),
+                ('city', '=', city),
+                ('state_id', '=', state),
+                ('country_id', '=', country),
+                ('type', '=', 'delivery'),
+                ('parent_id', '=', parent_id.id),
+            ]
+            existing_partner = self.env['res.partner'].search(domain, limit=1)
+            if existing_partner:
+                return existing_partner
+            else:
+                return self.env['res.partner'].create({
+                    'parent_id': parent_id.id,
+                    'type': 'delivery',
+                    'name': name,
+                    'street': street,
+                    'city': city,
+                    'state_id': state,
+                    'zip': zip_code,
+                    'country_id': country,
+                    'email': email,
+                    'phone': phone,
+                    'is_pickup_location': is_pickup_location,
+                    'location_data': location_data,
+                })
+        return False
```

Comment:
To align return type
```suggestion
        return self.env['res.partner']
```